06_iito/index.tex

```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
 
-Islands are also able to offer gifts without a request being made, meaning they can reach out to other islands if they want to boost their popularity.
\ No newline at end of file
+Islands are also able to offer gifts without a request being made, meaning they can reach out to other islands if they want to boost their popularity.
+
+\section{Implementation}
+\label{sec:IITO:Implementation}
+
+\subsection{Server Client Infrastructure}
+\label{subsec:IITO:server_client_infrastructure}  
+
+All IITO sessions, such as Gift Giving or Common Pool Donations, follow the same client server interaction format. The server acts a middleman for all agent to agent communication, and prompts agents to make decisions or formulate messages during the session. At each stage in a session the server compiles a list of messages from the agents, following that sessions' protocols, and then uses that list as input for the next stage. While a bit abstract this is better explained through an example.
+
+\subsection{Gifting Session}
+\label{subsec:IITO:gifting_session} 
+
+There are four phases to the gifting sessions: Gift requests, Gift offers, Gift responses and Updating gift history. Each part is further explained below and Figure~\ref{fig:IITO:gifting_session_diagram} shows the interactions a single client will have with the server over the session.
+\begin{itemize}
+    \item \textbf{Gift Requests:} Collects the gift requests of all islands. These requests include an amount the client wants and who they wish to request from.
+    \item \textbf{Gift Offers:} Gives the clients any requests that are directed to them and then collects the gift offers from each client. These offers include an amount the client wished to gift and to whom the gift is being offered.
+    \item \textbf{Gift Responses:} Provides the client with all the offers directed towards them and prompts them to respond to each of these offers. These responses contain the amount the client wishes to accept and a reason for their decision.
```

Comment:
again,

clients --> islands.

**It provides .**..  --> do NOT start with the verb directly

also put the semicolon out of \textbf{}
(not gonna repeat the same pattern of errors every time :) )